core/slice.go

```diff
@@ -325,9 +325,8 @@ func (sl *Slice) asyncPendingHeaderLoop() {
 
 // Read the phCache
 func (sl *Slice) readPhCache(hash common.Hash) (types.PendingHeader, bool) {
-	sl.phCachemu.RLock()
-	defer sl.phCachemu.RUnlock()
-	ph, exists := sl.phCache[hash]
+	phInt, exists := sl.phCache.Get(hash)
+	ph, _ := phInt.(types.PendingHeader)
```

Comment:
this will panic if the type assertion fails (e.g. if it didn't exist). Best to catch the failure instead, and only do it after checking `exists`:

```
// Read the phCache
func (sl *Slice) readPhCache(hash common.Hash) (types.PendingHeader, bool) {
	ph, exists := sl.phCache.Get(hash)
	if exists {
		if ph, ok := ph.(types.PendingHeader); ok {
			return ph, true
		}
	}
	return types.PendingHeader{}, false
}
```